### PR TITLE
Fix: fix submenu customized menuitem click crash issue

### PR DESCRIPTION
**Pull request by junhao3268**
# IMPORTANT

**Currently, this project is closed to any external contributions. Any pull request made against this project from external sources will likely be closed. If you would like to make changes to this project, please fork this project.**

# Guide

**This "Help" section can be deleted before submitting this pull request.**

*Update the name of this pull request to reflect the following shape:*

```
{type}/{scope?}/{message}
```

* **type** - A [conventional commit type](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional#type-enum) **REQUIRED**
* **scope** - The kabab-case scope of the changes in this request
* **message** - A short, kebab-case statement describing the changes **REQUIRED**

*Provide a general summary of the scope of the changes in this pull request.*

* [Readme](https://github.com/momentum-design/momentum-react-v2/blob/master/README.md)
* [Getting Started Guide](https://github.com/momentum-design/momentum-react-v2/blob/master/GETTING_STARTED.md)
* [Contributing Guide](https://github.com/momentum-design/momentum-react-v2/blob/master/CONTRIBUTING.md)

# Description
1.add title prop in ListItemBase for link button extra tooltip indicator.
2.fix submenu customized menuitem click crash issue.

image
image

*The full description of the changes made in this request.*

# Links

*Links to relevent resources.*

### CODE REVIEW

File: src/components/ListItemBase/ListItemBase.constants.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ const DEFAULTS = {
   ROLE: 'listitem',
   SHOULD_ITEM_FOCUS_BE_INSET: false,
   INTERACTIVE: true,
+  TITLE: 'title',
```
Comment (maxinteger) [inline src/components/ListItemBase/ListItemBase.constants.ts:23]: List item should not have default title and it never user anyway, please remove this line.

File: src/components/ListItemBase/ListItemBase.tsx [hunk 2]
```diff
@@ -288,6 +289,7 @@ const ListItemBase = (props: Props, providedRef: RefOrCallbackRef) => {
         data-interactive={interactive}
         className={classnames(className, STYLE.wrapper, { active: isPressed || isSelected })}
         role={role}
+        title={title}
```
Comment (maxinteger) [inline src/components/ListItemBase/ListItemBase.tsx:292]: I do not understand why do you use `title` attribute here. And looks like it does not related to the title of this PR. Could you please share the Jira issue link?

**Review by maxinteger — CHANGES_REQUESTED** [review]

File: src/components/ListItemBase/ListItemBase.tsx (hunk 2 above)
Comment (junhao3268) [reply src/components/ListItemBase/ListItemBase.tsx:292]: https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-505149
we discussed with @nataliadelmar about the visual indicator in last catchups, decide to add Icon after the list item (link like a button), and add a extra title for hover tooltip to further display the visible indicator, so we need to add the titlt prop to solve the type error.
image

Comment (junhao3268) [reply src/components/ListItemBase/ListItemBase.tsx:292]: > I do not understand why do you use `title` attribute here. And looks like it does not related to the title of this PR. Could you please share the Jira issue link?

We thought this part was also related to link bucket, so we put it in this pr, we should move this part to another PR.

**Review by maxinteger — APPROVED** [review]

File: src/components/ListItemBase/ListItemBase.tsx (hunk 2 above)
Comment (maxinteger) [reply src/components/ListItemBase/ListItemBase.tsx]: @junhao3268 yes, even if it is the same bucket it should be separated PR if it is not related to the original problem which is "fix submenu customized menuitem click crash issue" in this case.

I see you already moved your the title changes. Thank you

**Review by maxinteger — CHANGES_REQUESTED** [review]
sorry I have to add the `validated` label first :)

**Review by maxinteger — APPROVED** [review]
Looks good!

**Pull request merged**